holoviews/streams.py

```diff
@@ -768,7 +768,7 @@ def hashkey(self):
         for p in self.parameters:
             pkey = (p.owner, p.name)
             pname = self._rename.get(pkey, p.name)
-            key = ' '.join([p.owner.name, pname])
+            key = ' '.join([p.owner.name or str(id(p)), pname])
```

Comment:
This does not fix when the names are the same. Should the key just be `str(id(p))` or something similar?

Reply:
Honestly, yes, that's probably preferable.